tiledb/sm/query/readers/result_tile.h

```diff
@@ -213,12 +214,35 @@ class ResultTile {
     /*     CONSTRUCTORS & DESTRUCTORS    */
     /* ********************************* */
     TileData(
-        void* fixed_filtered_data,
-        void* var_filtered_data,
-        void* validity_filtered_data)
-        : fixed_filtered_data_(fixed_filtered_data)
-        , var_filtered_data_(var_filtered_data)
-        , validity_filtered_data_(validity_filtered_data) {
+        std::tuple<void*, ThreadPool::SharedTask> fixed_filtered_data,
+        std::tuple<void*, ThreadPool::SharedTask> var_filtered_data,
+        std::tuple<void*, ThreadPool::SharedTask> validity_filtered_data,
+        shared_ptr<FilteredData> filtered_data)
+        : fixed_filtered_data_(std::get<0>(fixed_filtered_data))
+        , var_filtered_data_(std::get<0>(var_filtered_data))
+        , validity_filtered_data_(std::get<0>(validity_filtered_data))
+        , fixed_filtered_data_task_(std::get<1>(fixed_filtered_data))
+        , var_filtered_data_task_(std::get<1>(var_filtered_data))
+        , validity_filtered_data_task_(std::get<1>(validity_filtered_data))
+        , filtered_data_(std::move(filtered_data)) {
+    }
+
+    ~TileData() {
+      try {
+        if (fixed_filtered_data_task_.valid()) {
+          auto st = fixed_filtered_data_task_.wait();
+        }
+
+        if (var_filtered_data_task_.valid()) {
+          auto st = var_filtered_data_task_.wait();
+        }
+
+        if (validity_filtered_data_task_.valid()) {
+          auto st = validity_filtered_data_task_.wait();
+        }
+      } catch (...) {
+        return;
```

Comment:
Each of these should be in its own try/catch.  If the first one throws then the other tasks would not be waited for.